api/operator/v1alpha1/certmanager_types.go

```diff
@@ -120,6 +128,30 @@ type CertManagerScheduling struct {
 	Tolerations []corev1.Toleration `json:"tolerations,omitempty" protobuf:"bytes,22,opt,name=tolerations"`
 }
 
+// Features contains experimental features.
+type Features struct {
+	// TechPreview is the list of features enabled under Technlogy Preview capabilities,
+	// is specific to the cert-manager Operator for Red Hat OpenShift and not in anyway
+	// related to the "TechPreviewNoUpgrade" feature set available in core OpenShift,
+	// although similar support and licensing terms may apply.
+	//
+	// Technology Preview features are not supported with Red Hat production service level agreements
+	// (SLAs) and might not be functionally complete. Red Hat does not recommend using them in production.
+	// These features provide early access to upcoming product features,
+	// enabling customers to test functionality and provide feedback during the development process.
+	//
+	// +kubebuilder:validation:XValidation:rule="oldSelf.size() != 0 ? oldSelf.all(e, self.exists(f, f == e)) : true",message="a tech preview feature once enabled, cannot be disabled."
+	// +kubebuilder:validation:XValidation:rule="self.all(e, self.filter(x, x == e).size() == 1)",message="tech preview features must not contain duplicate values."
+	// +kubebuilder:validation:Required
+	// +required
+	TechPreview TechPreview `json:"techPreview"`
+}
+
+// TechPreview contains the list of enabled experimental features.
+// +listType=atomic
+// +kubebuilder:validation:MaxItems=64
+type TechPreview []FeatureName
```

Comment:
nit: Wouldn't a struct with feature names as boolean type members be simple and CEL validations too would be less complex?
```
type Features struct {
   IstioCSR bool
}
```

Reply:
yes, initially I came up with that but the CEL validation(s) in Kubernetes seem to not allow validating the way I did. There's no way to do a general "check all fields" operation as a part of Kubernetes' security model where validations need to be always explicit.

The array option allows for simpler checking without needing to be explicit.

Reply:
CEL validation like:  `oldSelf.all(field, has(self[field]))` doesn't work, which ultimately lets the user unset a feature after it has been enabled.

Also, it seemed more tedious to write an expression that does equivalent of the aforesaid: 
```cel
!has(oldSelf) || (
    has(oldSelf.istioCSR) ? has(self.istioCSR) : true &&
    has(oldSelf.newFeature1) ? has(self.newFeature1) : true &&
    has(oldSelf.newFeature2) ? has(self.newFeature2) : true
    // ...
)
```
for each and every feature we add.